llvm/lib/Target/AIE/AIEBaseAddrSpaceInfo.h

```diff
@@ -0,0 +1,44 @@
+//===-- AIEBaseAddrSpaceInfo.h - Define Base AddressSpace Class  -*- C++-*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// (c) Copyright 2024 Advanced Micro Devices, Inc. or its affiliates
+//
+//===----------------------------------------------------------------------===//
+//
+// This file declares the AIEngine Base Address Space class.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_SUPPORT_AIEBASEADDRSPACEINFO_H
+#define LLVM_SUPPORT_AIEBASEADDRSPACEINFO_H
+
+#include <cstdint>
+
+namespace llvm {
+
+using MemoryBankBits = uint64_t;
+
+class AIEBaseAddrSpaceInfo {
+public:
+  virtual MemoryBankBits getDefaultMemoryBank() const {
+    // By default assume conflicts.
+    return ~0;
+  }
+  virtual MemoryBankBits
+  getMemoryBanksFromAddressSpace(unsigned AddrSpace) const {
+    // By default assume conflicts.
+    return ~0;
+  }
+
+  virtual bool addrspacesMayAlias(unsigned AS1, unsigned AS2) {
```

Comment:
I guess you will use that for `bool TargetTransformInfo::addrspacesMayAlias(unsigned AS0, unsigned AS1) const;`? How do you intend to get the right `AIEAddrSpaceInfo` for the current target?

Reply:
This is how I plan to get the AddrSpaceInfo for the correct target 

```
bool AIE2TTIImpl::addrspacesMayAlias(unsigned AS0, unsigned AS1) const {
  const AIEBaseAddrSpaceInfo &ASI = ST->getAddrSpaceInfo();
  return ASI.getMemoryBanksFromAddressSpace(AS0) &&
         ASI.getMemoryBanksFromAddressSpace(AS1);
}
```

`getAddrSpaceInfo` is defined in `llvm/lib/Target/AIE/AIE2Subtarget.cpp`

Reply:
Then could we remove `AIEBaseAddrSpaceInfo::addrspacesMayAlias`? Or do you plan to use it elsewhere? It seems to do the same as `AIE2TTIImpl::addrspacesMayAlias`.

Also, could `AIE2TTIImpl::addrspacesMayAlias` actually be `AIEBaseTTIImple::addrspacesMayAlias`? (if that even exists)

Reply:
1. Have removed `AIEBaseAddrSpaceInfo::addrspacesMayAlias`
2. No since we do not have `AIEBaseTTIImpl`, also based on discussion with Andrew the idea was scrapped https://github.com/Xilinx/llvm-aie/pull/173#discussion_r1763406439